### DISCUSSION TITLE
User/Accounts can enter their own OpenAI API key.

### DISCUSSION THREAD

**Discussion by mushroomsouper** [open-webui/open-webui#459] [Ideas]
I have ollama web-ui running and think it's great. My machine is too slow to run local models, but I love it as an interface for Chat-GPT and have cancelled my Plus subscriptions in favor of using OpenAI API in ollama. However it would be nice if users/accounts could setup their own Open AI API keys (rather than one key for every account to share). The use-case is: I could create myself a "work" account and use the company's OpenAI key for usage that is related to work.

**Comment by justinh-rahb** [comment]
This was the prior behaviour of the application before PR #381 . I have one user among my team that also misses the ability to set his own OpenAI endpoint/credentials in the frontend, but as an administrator I much prefer setting this centrally for everyone. I'm unsure of the feasibility of accomodating both methods, it seems it would lead to a lot of duplicated functionality between frontend and backend.

**Reply by mushroomsouper** [reply]
I can't speak to the feasibility or complexity, but how I imagine it would look is: 

Admin can set 'default' OpenAI key. Users then have a box they can check to enter and use their own key, that would override the default. OR, The default key appears to the user's settings as a disabled text input, they can click a button to make that field editable, they can then replace the default key with their own key. 

I'm not sure if that is any use to anyone, but definitely mark me down as someone who would like to see that kind of functionality.

**Reply by ITOTI-Y** [reply]
I also hope to add the function of allowing users to customize their API keys. Administrators can set a universal API key on the backend, so that any user can use that key to access the models. Meanwhile, specific users (such as paid users) can provide their own independent API keys to enable access to more models, which also makes it convenient for administrators to manage.

**Reply by justinh-rahb** [reply]
@ITOTI-Y This Function does exactly this: https://openwebui.com/f/hub/openai_manifold/

**Reply by ITOTI-Y** [reply]
> @ITOTI-Y This Function does exactly this: https://openwebui.com/f/hub/openai_manifold/该函数的作用正是如此： https://openwebui.com/f/hub/openai_manifold/
This is the feature I was looking for. Thank you very much.

**Reply by ShanySky** [reply]
> @ITOTI-Y This Function does exactly this: https://openwebui.com/f/hub/openai_manifold/此功能正是实现这一点: https://openwebui.com/f/hub/openai_manifold/

Can you show me how to use it?

**Reply by ijk0** [reply]
> > @ITOTI-Y This Function does exactly this: [https://openwebui.com/f/hub/openai_manifold/此功能正是实现这一点](https://openwebui.com/f/hub/openai_manifold/%E6%AD%A4%E5%8A%9F%E8%83%BD%E6%AD%A3%E6%98%AF%E5%AE%9E%E7%8E%B0%E8%BF%99%E4%B8%80%E7%82%B9): https://openwebui.com/f/hub/openai_manifold/
> 
> Can you show me how to use it?

to use this function, 
1. add the function in the admin pannel -> functions
2. in chat page, chat controls -> tool/function, select your function and input your apikey.

**Reply by ShanySky** [reply]
@ijk0 Thank you!
